src/traces/scattergl/index.js

```diff
@@ -8,29 +8,1026 @@
+        di.hts = Array.isArray(hoverlabel.font.size) ? hoverlabel.font.size[id] : hoverlabel.font.size;
+        di.htc = Array.isArray(hoverlabel.font.color) ? hoverlabel.font.color[id] : hoverlabel.font.color;
+        di.htf = Array.isArray(hoverlabel.font.family) ? hoverlabel.font.family[id] : hoverlabel.font.family;
+        di.hnl = Array.isArray(hoverlabel.namelength) ? hoverlabel.namelength[id] : hoverlabel.namelength;
+    }
+    var hoverinfo = trace.hoverinfo;
+    if(hoverinfo) {
+        di.hi = Array.isArray(hoverinfo) ? hoverinfo[id] : hoverinfo;
+    }
+
+    var fakeCd = {};
+    fakeCd[pointData.index] = di;
+
+    Lib.extendFlat(pointData, {
+        color: getTraceColor(trace, di),
+
+        x0: xc - rad,
+        x1: xc + rad,
+        xLabelVal: di.x,
+
+        y0: yc - rad,
+        y1: yc + rad,
+        yLabelVal: di.y,
+
+        cd: fakeCd
+    });
+
+    if(di.htx) pointData.text = di.htx;
+    else if(trace.hovertext) pointData.text = trace.hovertext;
+    else if(di.tx) pointData.text = di.tx;
+    else if(trace.text) pointData.text = trace.text;
+
+    fillHoverText(di, trace, pointData);
+    ErrorBars.hoverInfo(di, trace, pointData);
+
+    return [pointData];
+};
+
+
+ScatterRegl.selectPoints = function select(searchInfo, polygon) {
+    var cd = searchInfo.cd,
+        selection = [],
+        trace = cd[0].trace,
+        stash = cd[0].t,
+        x = stash.x,
+        y = stash.y;
+
+    var scene = stash.scene;
+
+    if(!scene) return selection;
+
+    var hasOnlyLines = (!subTypes.hasMarkers(trace) && !subTypes.hasText(trace));
+    if(trace.visible !== true || hasOnlyLines) return selection;
+
+    // degenerate polygon does not enable selection
+    // filter out points by visible scatter ones
+    var els = null, unels = null, i;
+    if(polygon !== false && !polygon.degenerate) {
+        els = [], unels = [];
+        for(i = 0; i < stash.count; i++) {
+            if(polygon.contains([stash.xpx[i], stash.ypx[i]])) {
+                els.push(i);
+                selection.push({
+                    pointNumber: i,
+                    x: x[i],
+                    y: y[i]
+                });
+            }
+            else {
+                unels.push(i);
+            }
+        }
+    }
+    else {
+        unels = Array(stash.count);
+        for(i = 0; i < stash.count; i++) {
+            unels[i] = i;
+        }
+    }
+
+    // create selection style once we have something selected
+    if(!scene.selectBatch) {
+        scene.selectBatch = Array(scene.count);
+        scene.unselectBatch = Array(scene.count);
+        scene.scatter2d.update(scene.unselectedOptions);
+    }
+    scene.selectBatch[trace.index || 0] = els;
+    scene.unselectBatch[trace.index || 0] = unels;
+
+    scene.clear();
+    scene.draw();
+
+    return selection;
+};
+
+
+ScatterRegl.style = function style() {};
```

Comment:
should be similar to `ScatterMapbox.style` here:

https://github.com/plotly/plotly.js/blob/master/src/traces/scattermapbox/index.js#L23